core/drivers/fabric-driver/server/events.ts

```diff
@@ -8,26 +8,43 @@ import ack_pb from '@hyperledger-labs/weaver-protos-js/common/ack_pb';
 import eventsPb from '@hyperledger-labs/weaver-protos-js/common/events_pb';
 import events_grpc_pb from '@hyperledger-labs/weaver-protos-js/relay/events_grpc_pb';
 import queryPb from '@hyperledger-labs/weaver-protos-js/common/query_pb';
-import { InteroperableHelper } from '@hyperledger-labs/weaver-fabric-interop-sdk'
-import { getDriverKeyCert } from "./walletSetup"
-import { DBConnector, LevelDBConnector } from "./dbConnector"
-import { checkIfArraysAreEqual, relayCallback } from "./utils"
+import { InteroperableHelper } from '@hyperledger-labs/weaver-fabric-interop-sdk';
+import { getDriverKeyCert } from "./walletSetup";
+import { DBConnector, LevelDBConnector } from "./dbConnector";
+import { checkIfArraysAreEqual, handlePromise, relayCallback } from "./utils";
+import { registerListenerForEventSubscription } from "./listener";
 
 const DB_NAME: string = "driverdb";
 
-function subscribeEventHelper(
+async function subscribeEventHelper(
     call_request: eventsPb.EventSubscription,
-    client: events_grpc_pb.EventSubscribeClient
+    client: events_grpc_pb.EventSubscribeClient,
+    network_name: string,
 ) {
     const newRequestId = call_request.getQuery()!.getRequestId();
-    addEventSubscription(call_request).then((requestId) => {
+    const [requestId, error] = await handlePromise(addEventSubscription(call_request));
+    if (error) {
+        const errorString: string = `error (thrown as part of async processing while storing to DB during subscribeEvent): ${JSON.stringify(error)}`;
+        console.error(errorString);
+        const ack_send_error = new ack_pb.Ack();
+        ack_send_error.setRequestId(newRequestId);
+        ack_send_error.setMessage(errorString);
+        ack_send_error.setStatus(ack_pb.Ack.STATUS.ERROR);
+        // gRPC response.
+        console.log(`Sending to the relay the eventSubscription error Ack: ${JSON.stringify(ack_send_error.toObject())}`);
+        // Sending the fabric state to the relay.
+        client.sendDriverSubscriptionStatus(ack_send_error, relayCallback);
+    } else {
         const ack_send = new ack_pb.Ack();
         if (newRequestId == requestId) {
             // event being subscribed for the first time
+
+            // the event listener logic follows here
+            await registerListenerForEventSubscription(call_request, network_name);
```

Comment:
need to put `handlePromise` on this function, and if there is an error in event subscription, need to send an error ack with an appropriate msg.
Success ack on Line 46, 47 should only be executed if register listener was successful.

Reply:
Done